{
  "id": "CVE-2019-16142",
  "assigner": "cve@mitre.org",
  "Published": "2019-09-09T10:15:10.96Z",
  "Modified": "2019-09-09T16:16:02.61Z",
  "last-modified": "2019-09-09T16:16:02.61Z",
  "summary": "An issue was discovered in the renderdoc crate before 0.5.0 for Rust. Multiple exposed methods take self by immutable reference, which is incompatible with a multi-threaded application.",
  "access": {
    "authentication": "NONE",
    "complexity": "LOW",
    "vector": "NETWORK"
  },
  "impact": {
    "availability": "PARTIAL",
    "confidentiality": "PARTIAL",
    "integrity": "PARTIAL"
  },
  "cvss": 7.5,
  "exploitabilityScore": 10.0,
  "impactScore": 6.4,
  "cvss-vector": "AV:N/AC:L/Au:N/C:P/I:P/A:P",
  "references": [
    "https://github.com/ebkalderon/renderdoc-rs/pull/32",
    "https://rustsec.org/advisories/RUSTSEC-2019-0018.html"
  ],
  "vulnerable_configuration": [
    "cpe:2.3:a:renderdocs-rs_project:renderdocs-rs:-:*:*:*:*:*:*:*",
    "cpe:2.3:a:renderdocs-rs_project:renderdocs-rs:0.1.0:*:*:*:*:*:*:*",
    "cpe:2.3:a:renderdocs-rs_project:renderdocs-rs:0.2.0:*:*:*:*:*:*:*",
    "cpe:2.3:a:renderdocs-rs_project:renderdocs-rs:0.3.0:*:*:*:*:*:*:*",
    "cpe:2.3:a:renderdocs-rs_project:renderdocs-rs:0.4.0:*:*:*:*:*:*:*"
  ],
  "vulnerable_product": [
    "cpe:2.3:a:renderdocs-rs_project:renderdocs-rs:-:*:*:*:*:*:*:*",
    "cpe:2.3:a:renderdocs-rs_project:renderdocs-rs:0.1.0:*:*:*:*:*:*:*",
    "cpe:2.3:a:renderdocs-rs_project:renderdocs-rs:0.2.0:*:*:*:*:*:*:*",
    "cpe:2.3:a:renderdocs-rs_project:renderdocs-rs:0.3.0:*:*:*:*:*:*:*",
    "cpe:2.3:a:renderdocs-rs_project:renderdocs-rs:0.4.0:*:*:*:*:*:*:*"
  ],
  "vendors": [
    "renderdocs-rs_project"
  ],
  "products": [
    "renderdocs-rs"
  ],
  "vulnerable_product_stems": [
    "cpe:2.3:a:renderdocs-rs_project:renderdocs-rs"
  ],
  "vulnerable_configuration_stems": [
    "cpe:2.3:a:renderdocs-rs_project:renderdocs-rs"
  ],
  "cwe": "CWE-20",
  "vulnerable_configuration_cpe_2_2": []
}